### PR TITLE
egrep: add page

### PR DESCRIPTION
**Pull request by antoine-amara**
add a dedicated page for egrep command. add 5 examples in this new page.

from the issue [https://github.com/tldr-pages/tldr/issues/6382](https://github.com/tldr-pages/tldr/issues/6382).

- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**

### CODE REVIEW

File: pages/common/egrep.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# egrep
+
+> Find patterns in files using extended regular expression (support `?`, `+`, `{}`, `()` and `|`).
```
Comment (marchersimon) [inline pages/common/egrep.md:3]:
```suggestion
> Find patterns in files using extended regular expression (supports `?`, `+`, `{}`, `()` and `|`).
```

File: pages/common/egrep.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+`egrep "{{search_pattern}}" {{path/to/file}}`
+
+- Search for a pattern within multiple files:
+
+`egrep "{{search_pattern}}" {{path/to/file1}} {{path/to/file2}} {{path/to/file3}}`
+
+- Find a pattern in a command result:
+
+`{{shell_command}} | egrep {{search_pattern}}`
```
Comment (marchersimon) [inline pages/common/egrep.md:16]: This is what we usually use for this
```suggestion
- Search stdin for a pattern:

`cat {{path/to/file}} | egrep {{search_pattern}}`
```

File: pages/common/egrep.md (hunk 1 above)
Comment (pixelcmtd) [reply pages/common/egrep.md:3]:
```suggestion
> Find patterns in files using extended regular expressions (supports `?`, `+`, `{}`, `()` and `|`).
```

**Comment by antoine-amara** [comment]
@marchersimon , @pixelcmtd , thanks for the review, I push a commit with some fix. I can squash these commits before merging if it is needed.

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**